project_template/inc/az/exampleshortname/operations.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/** @file operations.h */
+#pragma once
+#include <stdint.h>
+#include <az/exampleshortname/export.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+/** @brief add two numbers together
+ *  @param[in] a

```

Comment:
>a [](start = 15, length = 1)

consider putting a description of the parameters (like type and purpose of the parameter)
eg: The first `uint32_t` to add.

Reply:
We should strive for having enough annotation for generating our documentation with doxygen. @danewalton, can you please paste a link for our generated doxygen documentation here as a illustration?

---
In reply to: [309387205](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309387205) [](ancestors = 309387205)

Reply:
[Example header](https://github.com/Azure/azure-ulib-c/blob/master/inc/ustream.h)
[Documentation for that file and repo](https://azure.github.io/azure-ulib-c/ustream_8h.html)

---
In reply to: [309420922](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r309420922) [](ancestors = 309420922,309387205)

Reply:
I would argue that this is pretty well covered by the @return value.

Reply:
possibly for this example but consider a function that returns some library return value like `FOO_RESULT`. You can use the doxygen tag `@return` to describe the type of the return and then `@retval` to describe the different values that could be returned. [Example](https://github.com/Azure/azure-ulib-c/blob/6240cb4b7a74562dcd08374b49ac605708bc023a/inc/ustream.h#L87)

Reply:
yes I think that's a good idea.

Do you want any action here ,for this specific function?

Reply:
nitpicky but formatting it like [this](https://github.com/Azure/azure-ulib-c/blob/master/inc/ustream.h) does makes it nicely readable and gives a good amount of information. so things like a newline between the different tags, detailing the type of the parameters and the return parameter. we also have things tab aligned between the different components like for the params, it would be `@param[in] <TAB> a <TAB> The first <tt>uint32_t</tt> to add`